### ISSUE TITLE
High 404 failure rate on HLS streams (can't get init id)

### ISSUE THREAD

**Issue by troykelly** [AlexxIT/go2rtc#1221]
When requesting a HLS stream, a little over 50% of requests end up failing with a 404.

Repeated requests will ultimately get the stream.

This occurs even if the stream is already active with other consumers.

The go2rtc log shows the below (with id changing to reflect the request)

I'm not sure how to get extra hls logging, 

Notes:
 * Running inside frigate
 * Version: 1.9.4
 * Using Traefik to restore the access Frigate removed go2rtc
 * May be related to https://github.com/AlexxIT/go2rtc/issues/1166

```bash
[hls] can't get init id=X6IWhIFV
```

## ffmpeg log

...is too long, so it's here https://gist.github.com/troykelly/f730ef06acffff311239e0109afab57b

**Comment by AlexxIT** [comment]
The stream should respond with keyframe in 3 seconds. It is possible to increase this timeout, but to a reasonable value.
https://github.com/AlexxIT/go2rtc/blob/3762bdbccd599e173abb068c2d6cbfe9c424b105/internal/hls/session.go#L89-L104

**Comment by troykelly** [comment]
What if there are other HLS stream consumers active? or is a new stream started for every consumer?

**Comment by AlexxIT** [comment]
Yes. Each HLS consumer has personal session, buffer, etc. This is worst protocol for real time streaming.
This protocol only make sense for serving static files with different bitrate.
